ext/dom/element.c

```diff
@@ -858,8 +960,54 @@ PHP_METHOD(DOMElement, setAttributeNS)
 	if (errorcode != 0) {
 		php_dom_throw_error(errorcode, stricterror);
 	}
+}
 
-	RETURN_NULL();
+/* https://dom.spec.whatwg.org/#dom-element-setattributens */
+static void dom_set_attribute_ns_modern(dom_object *intern, xmlNodePtr elemp, zend_string *uri, zend_string *name, const char *value)
```

Comment:
```suggestion
static void dom_set_attribute_ns_modern(dom_object *intern, xmlNodePtr elemp, const zend_string *uri, const zend_string *name, const char *value)
```
Or is this not possible?

Reply:
Not entirely sure without trying it (I'm just commenting right now). Might not be possible because the underlying Zend APIs that get called actually _do_ make modifications (e.g. compute hash value).

Reply:
Ah right

Reply:
Yeah I could not do it for one of the arguments for that reason; did it for the other one tho.